pkg/volume/flexvolume/mounter.go

```diff
@@ -65,6 +65,13 @@ func (f *flexVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 
 	extraOptions := make(map[string]string)
 
+	// pod metadata
+	extraOptions[optionKeyPodName] = f.podName
+	extraOptions[optionKeyPodNamespace] = f.podNamespace
+	extraOptions[optionKeyPodUID] = string(f.podUID)
+	// service account metadata
+	extraOptions[optionKeyServiceAccountName] = f.podServiceAccountName
```

Comment:
> if we pass service account, should we also pass the secret too?

No, the point is not to let the volume plugin act as the service account, it is to inform the volume plugin *which* service account the pod is using. Think of a vault plugin that injects credentials from vault for that service account.